go-controller/pkg/ovn/master.go

```diff
@@ -1491,3 +1499,233 @@ func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
 	}
 	return nil
 }
+
+func (oc *Controller) addNodeObj(node *kapi.Node) error {
+	var gatewaysFailed sync.Map
+	var mgmtPortFailed sync.Map
+	var addNodeFailed sync.Map
+	var nodeClusterRouterPortFailed sync.Map
```

Comment:
In the original code, I see that these sync.Maps are defined once in:
```
func (oc *Controller) WatchNodes() {
	var gatewaysFailed sync.Map
	var mgmtPortFailed sync.Map
	var addNodeFailed sync.Map
	var nodeClusterRouterPortFailed sync.Map
```
... and then they are accessed, possibly concurrently, by the ResourceEventHandlerFuncs defined soon afterwards. 

Here we're redefining these sync.Maps locally in each `addNodObj`, `updateNodeObj` and deleteNodeObj`. Is the result equivalent?I have zero experience with `sync.Map`, so this is more a question for my knowledge than for this PR... :)

Reply:
I think you are right its up to k8s how it invoke those handlers and they could be concurrent that is probably the reason for those syncMap orginally I will fix my changes